chain/network/src/peer_manager.rs

```diff
@@ -369,13 +706,57 @@ impl PeerManagerActor {
             },
         );
 
-        self.process_edges(ctx, vec![new_edge.clone()]);
+        self.add_verified_edges_to_routing_table(ctx, vec![new_edge.clone()]);
+
+        checked_feature!(
+            "protocol_feature_routing_exchange_algorithm",
+            RoutingExchangeAlgorithm,
+            peer_protocol_version,
+            {
+                let addr2 = addr.clone();
+                near_performance_metrics::actix::run_later(
+                    ctx,
+                    file!(),
+                    line!(),
+                    WAIT_FOR_SYNC_DELAY,
+                    move |act, ctx2| {
+                        if peer_type == PeerType::Inbound {
+                            act.ibf_routing_pool
+                                .send(IbfRoutingTableExchangeMessages::AddIbfSet(peer_id, None))
+                                .into_actor(act)
+                                .map(move |response, act2, _ctx| {
+                                    match response {
+                                    Ok(IbfRoutingTableExchangeMessagesResponse::AddIbfSetResponse {
+                                        seed,
+                                        ibf_vec: _,
+                                    }) => act2.start_routing_table_syncv2(addr2, seed),
+                                    _ => error!(target: "network", "expected AddIbfSetResponse"),
+                                }
+                                })
+                                .spawn(ctx2);
+                        }
+                    },
+                );
+                self.send_sync(peer_type, addr, ctx, target_peer_id.clone(), new_edge, Vec::new());
+                return;
+            }
+        );
 
-        // TODO(MarX, #1363): Implement sync service. Right now all edges and known validators
-        //  are sent during handshake.
+        // TODO: This operation is expensive. Move creating `known_edges` to another thread.
```

Comment:
Please create an issue and add the issue number here

Reply:
Added